### PR TITLE
[TI_Anomali] Fix destination fields mapping conflicts

### PR DESCRIPTION
**Pull request by kcreddy**
## What does this PR do?

Field mapping conflicts exists between source and destination indices (created by transform) of Anomali. The issue happened because not all the files in the `elasticsearch/transform/latest_ioc/fields` folder is considered as destination fields. Only the fields inside the `fields.yml` are taken into the mapping. Hence dynamic mapping is being applied for rest of the fields.
   - This PR removes the additional files from `elasticsearch/transform/latest_ioc/fields` folder and and copies corresponding fields into `elasticsearch/transform/latest_ioc/fields/fields.yml`

**NOTE:** Due to issue in `elastic-package` not expanding the ECS fields under `elasticsearch/transform/latest_ioc/fields/fields.yml` mapped with `- external: ecs` definition, the ECS fields must be manually copied from `build/packages/ti_anomali/<VERSION>/data_stream/threatstream/fields/ecs.yml`
Reference Issue - https://github.com/elastic/elastic-package/issues/1369

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

**Comment by andrewkroh** [comment]
> Due to issue in elastic-package not expanding the ECS fields

Can you please link to the associated elastic-package bug issue in the PR description.

**Comment by kcreddy** [comment]
Added issue to the description

**Comment by kcreddy** [comment]
> IMO the package-spec is clear that it should honor any fields in the fields/ directory not just fields.yml

Yeah, I want to test it again to see if the problem is only the defined external fields are not getting expanded, or if multiple files not being honoured. I will update here.

File: packages/ti_anomali/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.14.1"
+  changes:
+    - description: Fix destination fields sort order and mapping
```
Comment (andrewkroh) [inline packages/ti_anomali/changelog.yml:4]: Somewhere (docs or changelog) I think we should mention that users should delete the logs-ti_anomali_latest.threatstream-1 after upgrading. (That is until we can automatically cleanup the old index).

Comment (kcreddy) [reply packages/ti_anomali/changelog.yml:4]: Makes sense.

Comment (kcreddy) [reply packages/ti_anomali/changelog.yml]: Updated the changelog to redirect to the docs. Added information about deleting older index and why we are doing it. Wording is generic enough to make users delete any future older versions. We could remove this doc section once the auto-cleanup is available.

**Comment by kcreddy** [comment]
> Does an issue already exist for elastic/kibana about Fleet only reading the contents of elasticsearch/transform/{name}/fields/fields.yml instead of elasticsearch/transform/{name}/fields/*?

I confirmed that multiple fields definition files inside transform folder is not working. I created the issue inside `elastic-package`: https://github.com/elastic/elastic-package/issues/1376

**Review by andrewkroh — APPROVED** [review]

**Pull request merged**